spec.md

```diff
@@ -3278,6 +3287,52 @@ Within a memory declaration, all port names must be unique.
 
 Any modifications to names must preserve the uniqueness of names within a namespace.
 
+# Intrinsics
+
+Intrinsics are expressions and statements which represent implementation-defined, compiler-provided functionality.
+
+Intrinsics generally are used for functionality which requires knowledge of the implementation or circuit not available to a library writer.
+Which intrinsics are supported by an implementation is defined by the implementation.
+
+Intrinsics first specify the intrinsic: the name of the intrinsic, any parameters, and return type if applicable.
+Any inputs to the intrinsic follow.
+An implementation shall type-check the specification and all operands.
+
+Below are some examples of intrinsics.
+These are for demonstration and their meaning or validity is determined by the implementation.
+
+The following shows an intrinsic expression for the intrinsic named "circt_ltl_delay" with two parameters, returns `UInt<1>`{.firrtl}, and has one operand.
+
+``` .firrtl
+FIRRTL version 4.0.0
+circuit Foo :
+  ;; snippetbegin
+  public module Foo :
+    input in : UInt<1>
+
+    node d = intrinsic(circt_ltl_delay<delay = 1, length = 0> : UInt<1>, in)
+  ;; snippetend
+```
+
+The following has an intrinsic statement with an intrinsic expression as its operand.
+The statement is for the intrinsic named "circt_verif_assert".
+The expression is for the intrinsic named "circt_isX" which returns a `UInt<1>`{.firrtl} and takes an operand.
+
+``` .firrtl
+FIRRTL version 4.0.0
+circuit Foo :
+  ;; snippetbegin
+  public module Foo :
+    input data : UInt<5>
+
+    intrinsic(circt_verif_assert, intrinsic(circt_isX: UInt<1>, data))
+  ;; snippetend
+```
+
+Operands and the return type of intrinsics must be passive and ground or aggregate types.
```

Comment:
This sentence suffers from ambiguity in boolean operator precedence.

Do you mean "passive and (ground or aggregate types)"?

Or do you mean "(passive and ground) or aggregate types"?

Reply:
Good call, thanks! Fixed!